openpype/pipeline/colorspace.py

```diff
@@ -118,6 +119,19 @@ def get_imageio_colorspace_from_filepath(
         if ext_match and file_match:
             colorspace_name = file_rule["colorspace"]
 
+    # if no file rule matched, try to get colorspace
+    # from filepath with OCIO v2 way
+    # QUESTION: should we override file rules from our settings and
+    #           in ocio v2 only focus on file rules set in config file?
```

Comment:
What if we just have a toggle in settings that toggles "Use OpenPype file rules" - when enabled it uses the file rules from settings, when disabled it uses the file rules from the OCIO file if it's OCIO v2?

It would potentially have use do much fewer compatibility checks. Also, it's best to assume `get_imageio_colorspace_from_filepath` will be called many times. I assume the compatibility check can be quite slow, especially if we're going through the subprocess wrapper so we might want to optimize it so we don't need to query it per file and maybe store it with the data?

Reply:
I wonder if you could cache the version compatibility check in the `CashedData` class. This way, the check would only be performed once, assuming the context is not changed to another configuration with a different version. However, I believe this could be an edge case that we don't need to worry about.

Reply:
@BigRoy look to ba237487a65176f2ee58e08533c65115b5a9c19d for more about the cashing of config version data.

Reply:
I could ofcourse make a current context of asset name for chacking of the cache validity.